docs/source/usersguide/settings.rst

```diff
@@ -235,6 +235,101 @@ file in your build directory. Setting the :attr:`openmc.Source.library`
 attribute to the path of this shared library will indicate that it should be
 used for sampling source particles at runtime.
 
+.. _serialized_custom_source:
+
+Custom Serialized Sources
+-------------------------
+
+If the custom source may be used with parameters at a variety of values then it
+may be necessary to serialize the source to an appropriate format in order to
+avoid recompiling the source library for each run. This is supported by defining
+a class inheriting from ``openmc::CustomSource`` that implements a
+``sample_source`` function. The class should also have logic to deserialise
+the parameters provided via the settings.xml file in the
+:attr:``openmc.Source.parameters`` attribute:
+
+.. code-block:: c++
+
+  #include <unordered_map>
+
+  #include "openmc/random_lcg.h"
+  #include "openmc/source.h"
+  #include "openmc/particle.h"
+
+  class SerializedSource : public openmc::CustomSource {
+    protected:
+      double energy_;
+
+      // Protect the constructor so that the class can only be created by serialisation.
+      SerializedSource(double energy) {
+        energy_ = energy;
+      }
+
+    public:
+      // Getters for the values that we want to use in sampling.
+      double energy() { return energy_; }
+
+      // Defines a function that can create a pointer to a new instance of this class
+      // by deserializing from the provided string.
+      static SerializedSource* from_string(const char* parameters) {
+        std::unordered_map<std::string, std::string> parameter_mapping;
+
+        std::stringstream ss(parameters);
+        std::string parameter;
+        while (std::getline(ss, parameter, ',')) {
+          parameter.erase(0, parameter.find_first_not_of(' '));
+          std::string key = parameter.substr(0, parameter.find_first_of('='));
+          std::string value = parameter.substr(parameter.find_first_of('=') + 1, parameter.length());
+          parameter_mapping[key] = value;
+        }
+
+        return new SerializedSource(std::stod(parameter_mapping["energy"]));
+      }
```

Comment:
I understand that this static method is here for deserialization purpose. I think for the sake of an example, it might be clearer to just put this logic in the `create` function, since its purpose is to take the string from parameters and return a pointer to a source. Also, I would consider just having parameters be the energy alone (i.e., `<parameters>1.0e6</parameters`) in this example to simplify the logic -- otherwise it can be easy to get lost in the details of string handling. A user can of course choose to do a "key=value, ...." format if they want, but if they just want one or two parameters, that's probably overkill.

Reply:
Makes sense for the documentation. Do you also have an opinion on the [runnable example](https://github.com/openmc-dev/openmc/blob/9c1b318e785b4ecbdbba9d58b50222e076cb50ec/examples/serialized_custom_source/serialized_source_ring.cpp)? Perhaps in that case it might make sense to give the user an idea of how they can pass in multiple values (and in an order independent way)?